app/views/forumthreads/index.html.erb

```diff
@@ -0,0 +1,84 @@
+<%= link_to "Forums", forums_path %> →
+<% if params.to_hash.slice("label", "title", "content", "author", "reply").size > 0 %>
+  <%= link_to "All Threads", forumthreads_path %> → Search Results
+<% else %>
+  <%= "All Threads" %>
+<% end %>
+<% params_list = params.to_hash.slice("id", "query", "label", "title", "content", "author", "reply") %>
+<h1>
+  <% if params[:id] %>
+      <% text = "forum '#{Forum.find(params[:id]).name}'" %>
+  <% else %>
+      <% text = "all threads" %>
+  <% end %>
+  <% if params_list.size > 0 %>
+    <%= title "Search results in #{text} (#{@threads.length})" %>
+  <% else %>
+    <% if params[:id] %>
+      <%= title "All threads in #{text}" %>
+    <% else %>
+      <%= title "All Threads" %>
+    <% end %>
+  <% end %>
+  <br>
+  <%= link_to "Advanced Search", search_forumthreads_path(params_list), class: "btn right blue" %>
+  <% if params_list.size > 0 && params[:id] %>
+    <%= link_to "Show All Threads", forumthreads_path(params_list.except("id")), class: "btn right blue" %>
+  <% elsif params_list.size > 0 && !params[:id] %>
+        <%= link_to "Show All Threads", forumthreads_path, class: "btn right blue" %>
+  <% end %>
+  <% if params[:id] %>
+    <%= link_to "Go to Forum", forum_path(params[:id]), class: "btn right blue" %>
+  <% end %>
+</h1>
+<br>
+<%= form_tag({controller: "forumthreads", action: "search_redirect"}, method: :post, style: "margin:0px;height:40px") do %>
+  <%= text_field_tag "query", nil, placeholder: "Search...", style: "margin:0px;height:40px;width:300px" %>
+  <% params.each do |key, value| %>
```

Comment:
Could you hardcode the few params actually used instead? Or use a fixed list of params, like `params_list`.